besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -695,6 +704,16 @@ public BesuController build() {
             ethProtocolManager,
             pivotBlockSelector);
 
+    ethPeers.setTrailingPeerRequirementsSupplier(synchronizer::calculateTrailingPeerRequirements);
+
+    if (SyncMode.isSnapSync(syncConfig.getSyncMode())
+        || SyncMode.isCheckpointSync(syncConfig.getSyncMode())) {
+      synchronizer.subscribeInSync((b) -> ethPeers.snapServerPeersNeeded(!b));
```

Comment:
` ethPeers.snapServerPeersNeeded` is called twice on L711 and L712, is it by design?